tests/components/aussie_broadband/test_config_flow.py

```diff
@@ -47,9 +49,59 @@ async def test_form(hass: HomeAssistant) -> None:
     }
     assert len(mock_setup_entry.mock_calls) == 1
 
+    # Test Already configured
+    result3 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    with patch("aussiebb.AussieBB.__init__", return_value=None), patch(
+        "aussiebb.AussieBB.get_services", return_value=fake_services
+    ), patch(
+        "homeassistant.components.aussie_broadband.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result4 = await hass.config_entries.flow.async_configure(
+            result3["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result4["type"] == RESULT_TYPE_ABORT
+
+
+async def test_no_services(hass: HomeAssistant) -> None:
+    """Test when there are no services."""
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result1["type"] == RESULT_TYPE_FORM
+    assert result1["errors"] is None
+
+    fake_services = []
+
+    with patch("aussiebb.AussieBB.__init__", return_value=None), patch(
+        "aussiebb.AussieBB.get_services", return_value=fake_services
+    ), patch(
+        "homeassistant.components.aussie_broadband.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_ABORT
+    assert len(mock_setup_entry.mock_calls) == 0
```

Comment:
https://github.com/nickw444/home-assistant/blob/d16c1350a5ac748b9a7c7acb709abfd2b6d5ec57/tests/components/aussie_broadband/test_config_flow.py#L156-L178

I added a test for this last night (forgot to push). I'll drop mine in a follow up commit, they are identical except for the assertion about `mock_setup_entry` calls and `assert result2["reason"] == "no_services_found"`